tests/scripts/thread-cert/Cert_6_1_02_REEDAttach_ED.py

```diff
@@ -0,0 +1,147 @@
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import unittest
+
+import config
+import mle
+import node
+
+LEADER = 1
+REED = 2
+MED = 3
+
+class Cert_6_1_2_REEDAttach_ED(unittest.TestCase):
+    def setUp(self):
+        self.simulator = config.create_default_simulator()
+
+        self.nodes = {}
+        for i in range(1,4):
+            self.nodes[i] = node.Node(i, (i == MED), simulator=self.simulator)
+
+        self.nodes[LEADER].set_panid(0xface)
+        self.nodes[LEADER].set_mode('rsdn')
+        self.nodes[LEADER].add_whitelist(self.nodes[REED].get_addr64())
+        self.nodes[LEADER].enable_whitelist()
+
+        self.nodes[REED].set_panid(0xface)
+        self.nodes[REED].set_mode('rsdn')
+        self.nodes[REED].add_whitelist(self.nodes[LEADER].get_addr64())
+        self.nodes[REED].add_whitelist(self.nodes[MED].get_addr64())
+        self.nodes[REED].enable_whitelist()
+        self.nodes[REED].set_router_upgrade_threshold(0)
+
+        self.nodes[MED].set_panid(0xface)
+        self.nodes[MED].set_mode('rsn')
+        self.nodes[MED].add_whitelist(self.nodes[REED].get_addr64())
+        self.nodes[MED].enable_whitelist()
+        self.nodes[MED].set_timeout(config.DEFAULT_CHILD_TIMEOUT)
+
+    def tearDown(self):
+        for node in list(self.nodes.values()):
+            node.stop()
+            node.destroy()
+        self.simulator.stop()
+
+    def test(self):
+        self.nodes[LEADER].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
+
+        self.nodes[REED].start()
+        self.simulator.go(5)
+        self.assertEqual(self.nodes[REED].get_state(), 'child')
+
+        self.nodes[MED].start()
+        self.simulator.go(config.DEFAULT_CHILD_TIMEOUT + 5)
+        self.assertEqual(self.nodes[MED].get_state(), 'child')
+        self.assertEqual(self.nodes[REED].get_state(), 'router')
+
+        med_messages = self.simulator.get_messages_sent_by(MED)
+
+        # Step 2 - DUT send MLE Parent Request
+        msg = med_messages.next_mle_message(mle.CommandType.PARENT_REQUEST)
+        self.assertEqual(0x02, msg.mle.aux_sec_hdr.key_id_mode)
```

Comment:
Good addition.
How about adding key_id_mode check to `check_parent_request() ` and using `check_parent_request() ` to replace below TLV checks?

Reply:
move `self.assertEqual(0x02, msg.mle.aux_sec_hdr.key_id_mode)` into `check_parent_request()`?
would it be better to add the key_id_mode check for all secured MLE message? as in Specification 7.3.1
> Secured MLE messages SHALL use only Key ID Mode 2